src/helpers/credential.rs

```diff
@@ -269,6 +279,138 @@ impl<'a> Credential<'a> {
         Ok(())
     }
 
+    pub async fn create_self_issued_credential(
+        &mut self,
+        schema_did: &str,
+        credential_subject_str: &str,
+        bbs_secret: &str,
+        bbs_private_key: &str,
+        credential_revocation_did: Option<&str>,
+        credential_revocation_id: Option<&str>,
+        exp_date: Option<&str>,
+    ) -> Result<String, CredentialError> {
+        let credential_subject: CredentialSubject = serde_json::from_str(credential_subject_str)?;
+        let credential_subject_copy: CredentialSubject = serde_json::from_str(credential_subject_str)?;
+        let subject_did = credential_subject.id.ok_or("no subject did found".to_string()).unwrap();
+        let subject_did_copy = subject_did.clone();
+        let subject_did_str = subject_did_copy.as_str();
+        let subject_did_str_copy = subject_did_str.clone();
+        let schema: CredentialSchema = self.get_did_document(schema_did).await?;
+        let issuer_public_key = self
+            .get_issuer_public_key(&subject_did, "#bbs-key-1")
+            .await?;
+        let issuer_public_key_copy= issuer_public_key.clone();
+        let use_valid_until= if exp_date.is_some() { true } else { false };
+        let exp_date_str = exp_date.unwrap_or("");
+
+        // need to add chrono crate
+        // let date_now = SystemTime::now();
+        // let datetime: DateTime<Utc> = date_now.into();
+        // let date_now_str = datetime.format("%d/%m/%Y %T");
+
+        // create credential offer
+        let unsigned_credential = UnsignedBbsCredential {
+            context: vec![
+                "https://www.w3.org/2018/credentials/v1".to_string(),
+                "https://schema.org/".to_string(),
+                "https://w3id.org/vc-revocation-list-2020/v1".to_string(),
+            ],
+            id: "uuid:834ca9da-9f09-4359-8264-c890de13cdc8".to_string(),
+            r#type: vec!["VerifiableCredential".to_string()],
+            issuer: subject_did,
+            valid_until: if use_valid_until {
+                Some(exp_date_str.to_string())
+            } else {
+                None
+            },
+            issuance_date: "2023-02-22T00:00:00.000Z".to_string(), //date_now_str.to_string(),
+            credential_subject: credential_subject_copy,
+            credential_schema: CredentialSchemaReference {
+                id: schema.id,
+                r#type: schema.r#type,
+            },
+            credential_status: CredentialStatus {
```

Comment:
Can we use the unused revocation parameters here?

Reply:
Done